packages/connect-react/src/index.ts

```diff
@@ -1,7 +1,139 @@
+  orgError: Error | null
+  orgLoading: boolean
+}
+
+type ConnectProps = {
+  children: React.ReactNode
+  connector: ConnectorDeclaration
+  location: string
+  options?: ConnectOptions
+}
+
+type OrganizationHookResult = [Organization | null, Error | null, boolean]
+type AppsHookResult = [Application[], Error | null, boolean]
+type PermissionsHookResult = [Permission[], Error | null, boolean]
+
+const ConnectContext = React.createContext<ContextValue>({
+  org: null,
+  orgError: null,
+  orgLoading: false,
+})
+
+export function Connect({
+  children,
+  connector,
+  location,
+  options,
+}: ConnectProps): JSX.Element {
+  const [org, setOrg] = useState<Organization | null>(null)
+  const [orgError, setOrgError] = useState<Error | null>(null)
+  const [orgLoading, setOrgLoading] = useState<boolean>(false)
+
+  useEffect(() => {
+    let cancelled = false
+
+    const update = async () => {
+      setOrgLoading(true)
+      try {
+        const org = await connect(location, connector, options)
+        if (!cancelled) {
+          setOrgError(null)
+          setOrg(org)
+        }
+      } catch (err) {
+        if (!cancelled) {
+          setOrgError(err)
+          setOrg(null)
+        }
+      } finally {
+        if (!cancelled) {
+          setOrgLoading(false)
+        }
+      }
+    }
+
+    update()
+
+    return () => {
+      cancelled = true
+    }
+  }, [location, connector, options])
+
+  const value = useMemo<ContextValue>(() => ({ org, orgError, orgLoading }), [
+    org,
+    orgError,
+    orgLoading,
+  ])
+
+  return React.createElement(ConnectContext.Provider, { value, children })
 }
 
-export function AragonConnector(): void {
-  return
+export function useOrganization(): OrganizationHookResult {
+  const { org, orgError, orgLoading } = useContext<ContextValue>(ConnectContext)
+  return [org, orgError, orgLoading]
 }
+
+function useConnectSubscription<Data>(
+  cb: (
+    org: Organization,
+    onData: (data: Data) => void
+  ) => { unsubscribe: Function },
+  initValue: Data
+): [Data, Error | null, boolean] {
+  const [data, setData] = useState<Data>(initValue)
+  const [loading, setLoading] = useState<boolean>(false)
+  const { org } = useContext<ContextValue>(ConnectContext)
+  const cbRef = useRef(cb)
+
+  useEffect(() => {
+    if (!org) {
+      return
+    }
+
+    let handler: any
+
+    const update = async () => {
+      setLoading(true)
+
+      handler = cbRef.current(org, (data: Data) => {
+        setData(data)
+        setLoading(false)
+      })
+    }
+    update()
+
+    return () => {
+      handler?.unsubscribe?.()
+    }
+  }, [org])
+
+  return [data, null, loading]
+}
+
+export function useApps(): AppsHookResult {
+  return useConnectSubscription<Application[]>(
+    (org, onData) => org.onApps(onData),
+    []
+  )
+}
+
+export function usePermissions(): PermissionsHookResult {
+  return useConnectSubscription<Permission[]>(
+    (org, onData) => org.onPermissions(onData),
+    []
+  )
+}
+
```

Comment:
I wonder if we should have a hook in the lines of `useIntent`. This was something that 1Hive used when working on the honey-pot MVP. You can take a look at the specific logic of how they handle this: [here](https://github.com/1Hive/honey-pot/blob/master/src/hooks/useActions.js#L92).

Reply:
As discussed offline: let’s use `appIntent()` in the examples first, and see if an opportunity to provide a hook appear.